admin/services/data-access/check-form.data.service.js

```diff
@@ -129,18 +129,28 @@ const checkFormDataService = {
       sortOrder = 'DESC'
     }
     const params = []
-    const sql = `SELECT cf.*, cw.[name] FROM [mtc_admin].[checkForm] cf
-    INNER JOIN [mtc_admin].[checkFormWindow] fw
-        ON cf.id = fw.checkForm_id
-    INNER JOIN [mtc_admin].[checkWindow] cw
-      ON cw.id = fw.checkWindow_id
-    WHERE cf.isDeleted=0 AND fw.checkWindow_id=@windowId
-    ORDER BY cw.[name] ${sortOrder}`
-    params.push({
-      name: 'windowId',
-      value: windowId,
-      type: TYPES.Int
-    })
+    let sql
+    if (windowId) {
+      sql = `SELECT cf.*, cw.[name] FROM [mtc_admin].[checkForm] cf
```

Comment:
There is no need to specify the schema here '[mtc_admin]'. Suggest we omit the schema.

Reply:
🤔  its fully qualified.  if there were other schemas in the same database, and they also contained a table of the same name it would fail

Reply:
Perhaps I don't understand what's going on here.  I assumed that it was connecting to the `mtc_admin` schema during setup, but looking at the code that doesn't seem to be the case.

Does that mean that ms-sql will look up the table name and choose some matching table from the information schema??  Not good!  So there is a performance hit in the lookup and the table may not actually even come from the correct schema...

Reply:
the connection connects to the database.  The database can contain many schemas.  Its standard practice to include the schema prefix in queries.  You *can* use the default `dbo` schema, but this is not good practice.  Regarding the performance hit - i can't find anything to clarify this either way, but given the user only has one schema it will be negligible.  Also, this is standard MSSQL stuff - it's no doubt heavily optimised from the perspective of object metadata etc.

Reply:
I agree (now) we should include the schema name.   I assumed the schema was part of the connection, as using an unspecified schema can't be good.

I would argue that the schema name is ostensibly  under the control of the DBA, so we should be able to change it with minimal effort - so we should use the variable exported from sqlService.schemaName instead of the hardcoded string.

I'm not arguing that strongly for this, as it's still fairly easy to do a search and replace, but the first solution is more elegant.

Reply:
agreed.